### PR TITLE
Add endof for Pair

### PR DESCRIPTION
**Pull request by joehuchette**
`first(::Pair)` already works on master, this adds the corresponding `last` method.

### CODE REVIEW

**Comment by mlubin** [comment]
Ref https://github.com/JuliaLang/julia/issues/12261 which broke JuMP

File: base/operators.jl [hunk 1]
```diff
@@ -441,6 +441,9 @@ getindex(p::Pair,i::Int) = getfield(p,i)
 getindex(p::Pair,i::Real) = getfield(p, convert(Int, i))
 reverse(p::Pair) = Pair(p.second, p.first)
 
+first(p::Pair) = p.first
```
Comment (mlubin) [inline base/operators.jl]: Why add this definition if ``first(::Pair)`` already works?

**Comment by joehuchette** [comment]
Alternatively, I think this could be implemented by adding ``endof(::Pair)``.

File: base/operators.jl (hunk 1 above)
Comment (joehuchette) [inline base/operators.jl]: Only for clarity

**Comment by mlubin** [comment]
@joehuchette, could you squash?

**Comment by mlubin** [comment]
Are there any objections to this? Since you can already do ``(:a => :b)[1]`` and ``(:a => :b)[2]``, we're just adding ``(:a => :b)[end]``.

**Comment by tkelman** [comment]
this lgtm, ci's a bit backed up since there were a lot of merges today

**Comment by jakebolewski** [comment]
Should help

Liver Salt

**Pull request merged**